### PR TITLE
Feature: Integrating settings page

### LINKED ISSUE #122: Integrating settings page
**Describe the task that needs to be done.**

The UI of the Settings page has been implemented in #83. However, it is yet to be integrated into the current React router so that it can be accessed from the `SideBar` component by the users. This issue should implement those requirements. 

**Describe how a solution to your proposed task might look like (and any alternatives considered).**

There would be changes made to the `SideBar` component to include a button that would take the user to the `Settings` page component. Thereafter, an `onEffect()` should be added to the `Settings` component so that it populates the initial state of the component based on what the current user settings are in the backend (we have to make a backend request here).

To change the `pairingEnabled` property for the user, we need to attach an event listener which triggers a backend request each time the button is toggled. These are the basic implementation details for this feature.

**Notes**

* The original assignee was unable to work on this issue during the weekend before submission, hence in the full team meeting on 20th March, it was decided that we should get this feature implemented ASAP to avoid any last-minute changes.
* This issue was approved by everyone in the meeting.

### PR DESCRIPTION
**Pull request by shreytailor**
## Description

The Settings page has now been integrated with the application, as its entry has been added to the `SideBar` component and a new protected route has been added to `/settings` which exposes the Settings component. Furthermore, the Settings component now uses helper functions from `UsersAPI` to create backend requests in order to toggle the `pairingEnabled` mode in the backend `User` entity.

Closes #122

## How has this been tested?

Manual testing has been performed using the Swagger UI implemented by the backend team.

## Screenshots

image

image

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/pages/Routes/Routes.jsx [hunk 1]
```diff
@@ -18,6 +19,7 @@ function Routes() {
         <Route path="/find-matches" element={<Pairing />} />
         <Route path="/chat" element={<Chat />} />
         <Route path="/accounts" element={<Example />} />
+        <Route path="/settings" element={<Settings />} />
```
Comment (scorpionknifes) [inline src/pages/Routes/Routes.jsx:22]: is there a difference between accounts and settings? also would it be good idea to remove accounts if we using settings instead?

Comment (shreytailor) [reply src/pages/Routes/Routes.jsx:22]: I thought it was a placeholder for something that's to be added. Shall I remove it then?

**Review by scorpionknifes — APPROVED** [review]

**Review by dshah1901 — APPROVED** [review]
LGTM !! Thanks!!!

File: src/pages/Routes/Routes.jsx (hunk 1 above)
Comment (shreytailor) [reply src/pages/Routes/Routes.jsx]: The `/settings` route has been removed from the router, as well as the `SideBar` component. The new screenshots have been added to the PR.

**Pull request merged**